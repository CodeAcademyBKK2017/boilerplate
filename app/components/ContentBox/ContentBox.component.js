import Icon from 'react-native-vector-icons/Foundation';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './ContentBox.style';
import {
  Text,
  TextInput,
  TouchableOpacity,
  View
} from 'react-native';

export default class ContentBox extends Component {

  render () {
    return (
      <View style={styles.contentStyle}>
        <View style={styles.contentStyleColum}>
          <Text style={styles.textContent}>Please type your note below.</Text><Icon name='clipboard-notes' size={20} color='#333333' />
        </View>
        <TextInput 
          style={styles.textInput}
          multiline= {true}
          value={this.props.contentValueText}
          placeholder={'- Understand how react-native works.\n- Build a native android and iOS app.\n- Setup CI for automated builds.\n- Conventions to manage the codebase.'}
          onChangeText={this.props.onContentChange}
          underlineColorAndroid='transparent'
        />
        <View style={styles.saveTabStyle}>
          <TouchableOpacity style={styles.flexOne} onPress={this.props.onSave}><Text style={styles.saveTextStyle}>Save</Text></TouchableOpacity>
          <Text>{this.props.count} chacters</Text>
        </View>
      </View>
    );
  }
}

ContentBox.propTypes = {
  onContentChange: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  contentValueText: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired
};

ContentBox.defaultProps = {
  onContentChange: PropTypes.func,
  onSave: PropTypes.func,
  contentValueText: '',
  count: 0
};