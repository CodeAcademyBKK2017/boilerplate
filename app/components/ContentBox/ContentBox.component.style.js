import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  contentStyle: {
    flex: 1,
    width: '100%',
    paddingLeft: 10,
    paddingRight: 10
  },
  contentStyleColum: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 15,
    paddingBottom: 15
  },
  textInput: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    borderColor: 'gray', 
    borderWidth: 1,
    borderRadius: 3,
    backgroundColor: 'white',
    padding: 10
  },
  textContent: {
    flex: 1,
    justifyContent: 'flex-start',
    fontStyle: 'italic',
    color: '#333333'
  },
  saveTabStyle: {
    flexDirection: 'row',
    paddingTop: 15,
    paddingBottom: 15
  },
  saveTextStyle: {
    fontWeight: 'bold'
  },
  flexOne: {
    flex: 1
  }
});
