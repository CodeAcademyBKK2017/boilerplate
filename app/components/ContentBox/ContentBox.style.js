import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  contentStyle: {
    flex: 1,
    width: '100%',
    paddingLeft: 10,
    paddingRight: 10
  },
  contentStyleColum: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop: 15,
    paddingBottom: 15
  },
  textInput: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    borderColor: 'gray', 
    borderWidth: 1,
    borderRadius: 3,
    backgroundColor: 'white',
    padding: 10
  },
  textContent: {
    justifyContent: 'flex-start',
    fontStyle: 'italic',
    color: '#333333',
    marginRight: 5
  },
  saveTabStyle: {
    flexDirection: 'row',
    paddingTop: 15,
    paddingBottom: 15
  },
  saveTextStyle: {
    fontWeight: 'bold'
  },
  flexOne: {
    flex: 1
  }
});
