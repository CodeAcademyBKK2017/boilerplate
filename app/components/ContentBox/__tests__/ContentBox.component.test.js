import 'react-native';
import ContentBox from '../ContentBox.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('ContentBox', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ContentBox />
    );
    expect(tree).toBeDefined();
  });

  it('ContentBox: MatchSnapshot', () => { // example to test class methods
    const tree = renderer.create(<ContentBox />);
    expect(tree).toMatchSnapshot();
  });
});