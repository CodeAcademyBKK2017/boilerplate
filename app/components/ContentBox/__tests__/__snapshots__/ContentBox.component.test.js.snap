// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox ContentBox: MatchSnapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingBottom": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "fontStyle": "italic",
          "justifyContent": "flex-start",
          "marginRight": 5,
        }
      }
    >
      Please type your note below.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "fontcustom",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï„¨
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    placeholder="- Understand how react-native works.
- Build a native android and iOS app.
- Setup CI for automated builds.
- Conventions to manage the codebase."
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 3,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "flex-start",
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingBottom": 15,
        "paddingTop": 15,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Save
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      0
       chacters
    </Text>
  </View>
</View>
`;
